server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureService.java

```diff
@@ -116,10 +123,15 @@ public SearchBackpressureService(
         );
 
         this.taskCancellationRatioLimiter.set(
-            new TokenBucket(state::getCompletionCount, getSettings().getCancellationRatio(), getSettings().getCancellationBurst())
+            new TokenBucket(this::getTaskCompletionCount, getSettings().getCancellationRatio(), getSettings().getCancellationBurst())
```

Comment:
This means that SearchShardTask and SearchTask cancellation limits are shared. Is this by design?

Reply:
Yes, we have kept the cancellation limits shared between both the tasks.

Reply:
Okay, got your point. Having separate limiters for SearchTask and SearchShardTask will give better control to the customer on how they want the cancellation to be done.
Will introduce new dynamic settings for cancellation rate, ration and burst for SearchTask as well.